jxl/src/frame/modular/tree.rs

```diff
@@ -173,6 +199,97 @@ pub(super) fn predict(
     }
 }
 
+/// Optimized prediction using flat tree (matches C++ context_predict.h:351-371).
+#[inline]
+#[allow(clippy::too_many_arguments)]
+pub(super) fn predict_flat(
```

Comment:
We should factor out all the code except the tree navigation itself to a single function shared between flat and non-flat tree prediction.

Reply:
Done. Factored out the property computation into a shared `compute_properties()` function that both `predict()` and `predict_flat()` now use.